In [6]:
import sys
sys.path.append("../src/")

In [8]:
%load_ext autoreload
%autoreload 2
import training
from training import ModelRunner

The autoreload extension is already loaded. To reload it, use:
  %reload_ext autoreload


In [108]:
args = "--epochs 2 --emsize 768 --nhid 1536 --bptt 10 --batch_size 4 --dropout 0 --nlayers 2 --log-interval 1 --lr 1"
runner = ModelRunner(args.split())

In [19]:
runner.evaluate()

Loading batch 0
Start model
Finish model
0.125 47.042755126953125
Loading batch 1
Start model
Finish model
0.05 51.24909210205078
Loading batch 2
Start model
Finish model
0.05 43.89897918701172
Loading batch 3
Start model
Finish model
0.125 53.84113693237305
Loading batch 4
Start model
Finish model
0.05 77.6212387084961
Loading batch 5
Start model
Finish model
0.05 93.9007339477539
Loading batch 6
Start model
Finish model
0.1 97.24732971191406
Loading batch 7
Start model
Finish model
0.125 54.366554260253906


KeyboardInterrupt: 

In [110]:
for i in range(20):
    runner.train(i, 5)

| epoch   0 |     1/28203 batches | lr 1.00000 | ms/batch 1490.51 | loss  2.57 | acc 0.85
| epoch   0 |     2/28203 batches | lr 1.00000 | ms/batch 692.04 | loss  1.36 | acc 0.775
| epoch   0 |     3/28203 batches | lr 1.00000 | ms/batch 819.91 | loss  1.44 | acc 0.775
| epoch   0 |     4/28203 batches | lr 1.00000 | ms/batch 516.21 | loss  2.22 | acc 0.65
| epoch   0 |     5/28203 batches | lr 1.00000 | ms/batch 661.87 | loss  3.94 | acc 0.7
| epoch   1 |     1/28203 batches | lr 1.00000 | ms/batch 1504.81 | loss  2.14 | acc 0.7375
| epoch   1 |     2/28203 batches | lr 1.00000 | ms/batch 684.60 | loss  2.78 | acc 0.625
| epoch   1 |     3/28203 batches | lr 1.00000 | ms/batch 787.46 | loss  2.29 | acc 0.675
| epoch   1 |     4/28203 batches | lr 1.00000 | ms/batch 525.48 | loss  2.96 | acc 0.525
| epoch   1 |     5/28203 batches | lr 1.00000 | ms/batch 664.02 | loss  3.46 | acc 0.575
| epoch   2 |     1/28203 batches | lr 1.00000 | ms/batch 1482.95 | loss  3.25 | acc 0.7875
| epoch  

KeyboardInterrupt: 

# Current Issues
* optimize batch sizes (check gpu usage)
* choose correct batch size, dropout, bptt, num of negatives, model size
* doesn't train better than loss .2 / ac ~87.5%

## Notes:
* slowest part is data loading at 0.07 sec
* 16 minutes per epoch (data to 5k)
* ACC starts to grow around loss 2 or 3
* lr = 10 might be good
* --bptt 256 --batch_size 16 --nlayers 2 works --- 85% GPU util
